modules/raffle.py

```diff
@@ -18,25 +20,30 @@ def install(self, application: Application) -> None:
         application.add_handler(CommandHandler("raffle_list", self.__raffle_list))
 
     async def __raffle(self, update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
-        username = update.message.from_user.username
+        user = update.message.from_user.username
+
+        try:
+            if not user:
+                raise UserFriendlyError('You first need to choose a username in Telegram')
 
-        # Process the username and send a reply
-        if username:
             try:
-                self.lc.remove_points(username, 5)
-                self.entries.append(username)
-                entry_count = Counter(self.entries)
-                reply_text = f"Congrats @{username}. You just bought a ticket for the Community Raffle. Thanks for supporting and good luck!\n\n"
-
-                # Display the list of entries with entry counts
-                for entry, count in entry_count.items():
-                    reply_text += f"@{entry} - {count} Ticket(s)\n"
-            except Exception as e:
-                reply_text = str(e)
-        else:
-            reply_text = "First, create a username in Telegram!"
-
-        await update.message.reply_text(reply_text)
+                self.lc.remove_points(user, Points(5))
+            except InsufficientFundsError as e:
+                raise UserFriendlyError(f"Oh no @{user}! You don't have enough points for the Community Raffle. Buy some drinks from the bar or beg a friend for a donation!") from error
```

Comment:
this will not interrupt the bot execution?

Reply:
There is indeed an issue here witht he naming of the e / error variable. It must've snuck in as I was doing merges. I pushed a fix just now.

As for the control flow itself, it won't interrupt the bot execution since the new exception is caught further below. This is all within a larger try/except block.